src/Segmentation/regionprops.jl

```diff
@@ -2,25 +2,376 @@
+    type_vals[[5, 7, 15, 17, 25, 27]] .= 1
+    type_vals[[21, 33]] .= sqrt(2)
+    type_vals[[13, 23]] .= (1 + sqrt(2)) / 2
+
+    # Convolution array for classifying boundary pixel type
+    conv_arr = [10 2 10; 2 1 2; 10 2 10]
+
+    results = conv(e, conv_arr; algorithm=:direct)
+
+    # Count instances of boundary types and multiply to get the perimeter
+    val_counts = Dict()
+    for val in vec(results)
+        val_counts[val] = get(val_counts, val, 0) + 1
+    end
+    perim = 0
+
+    for val in keys(val_counts)
+        (val == 0 || val > 33) && continue
+        perim += type_vals[val] * val_counts[val]
+    end
     
+    return perim
+end
+
+
+"""component_convex_area(A; algorithm=PixelConvexArea()")
+
+Compute the convex area of labeled regions. Two methods available: "PixelConvexArea" and "PolygonConvexArea".
+The polygon method uses Green's theorem to find the area of a polygon through its line integral, 
+while the pixel method uses a point-in-pixel calculation to determine if pixels are inside the
+convex hull. In general the polygon area will be smaller than the pixel area.
+"""
+function component_convex_area(A;
+    algorithm::ConvexAreaEstimationAlgorithm=PixelConvexArea()
+    ) 
+    mn, mx = extrema(A)
+    if !(mn == 0 || mn == 1)
+        throw(ArgumentError("The input labeled array should contain background label `0` as the minimum value"))
+    end
+    return algorithm(A)
+end
+
+"""PolygonConvexArea(minimum_area=4)
+
+Estimate the convex area by integrating the area of the convex hull polygon.
+Uses the convexhull function from ImageMorphology, which raises an error if the
+area of the segment is less than or equal to 3. In general, the error should be smaller
+for larger shapes.
+"""
+@kwdef struct PolygonConvexArea <: ConvexAreaEstimationAlgorithm
+    minimum_area = 4
+end
+
+function (f::PolygonConvexArea)(A)
+    mn, mx = extrema(A)
+    areas = component_lengths(A)
+
+    convex_areas = zeros(Float64, 0:mx)
+    for i in unique(A)
+        # treat convex area background and too-small objects as undefined
+        (i == 0) || (areas[i] < f.minimum_area) && begin
+            convex_areas[i] = NaN
+            continue
+        end
+
+        chull = convexhull(A .== i)
+        N = length(chull)
+
+        ca = 0
+        for j in 1:N
+            x0, y0 = Tuple(chull[j])
+            x1, y1 = Tuple(chull[(j % N) + 1])
+            ca += x0*y1 - y0*x1
+        end
+        ca *= 0.5
+        convex_areas[i] = ca
+    end
+    return convex_areas
+end
+
+
+"""PixelConvexArea()
+
+Estimate the convex area by integrating the area of the convex hull polygon.
+Uses the convexhull function from ImageMorphology, which raises an error if the
+area of the segment is less than or equal to 3. In general, the error should be smaller
+for larger shapes.
+"""
+@kwdef struct PixelConvexArea <: ConvexAreaEstimationAlgorithm
+    minimum_area = 4
+end
+
+function (f::PixelConvexArea)(A)
+    _, mx = extrema(A)
```

Comment:
If only the max is need consider using `maximum`.
```suggestion
    mx = maximum(A)
```

Reply:
Fair, this was mirroring the other `component_*` functions from image morphology and used the minimum for checking if the background was 0. I dropped the mn since it wasn't used in that section.